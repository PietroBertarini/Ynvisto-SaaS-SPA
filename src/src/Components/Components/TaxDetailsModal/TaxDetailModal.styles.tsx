import styled from 'styled-components';

export const ModalDetails = styled.div`
  background-color: #1E1E23;


  h1 {
    color: #fff;
    font-size: 2vw;
    text-align: center;
    border-bottom: 1px solid #E9C46A;
    width: 100%;
  }

  .operationType {
    h2 {
      font-size: 1.5vw;
      color: #fff;
      text-align: center;
      margin-bottom: 1;
      margin-top: 3;
    }

    .taxes {
      display: flex;
      flex-direction: row;  
      justify-content: space-between;

      p {
        font-size: 0.8vw;
        color: #fff;
        margin-top: 1;
      }
    }

    .result {
      display: flex;
      flex-direction: row;  
      justify-content: space-between;

      p {
        font-size: 0.8vw;
        color: #E9C46A;
        margin-top: 1;
      }
    }

    .line {
      border-bottom: 1px solid #fff;
      width: 100%;
    }
  } 
`;
