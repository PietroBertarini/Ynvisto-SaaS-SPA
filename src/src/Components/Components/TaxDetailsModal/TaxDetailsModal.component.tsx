import React from 'react';
import { ModalDetails } from './TaxDetailModal.styles';
import { ITaxdDetailInfo } from '../MonthlyCardReport/MonthlyCardReport.utils';

interface OwnProps {
  data: ITaxdDetailInfo,
}

export default function TaxDetailsModal(props: OwnProps) {
  const { data } = props;

  return (
    <ModalDetails>
      <h1>Detalhamento</h1>

      {data.commonOps && (
      <div className="operationType">
        <h2>Operações Comuns</h2>
        <div className="taxes">
          <p>Venda total do mês</p>
          <p>
            {data.commonOps.totalSell.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Resultado do mês</p>
          <p>
            {(data.commonOps.monthProfit >= 0) && '+'}
            {data.commonOps.monthProfit.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Lucro tributável</p>
          <p>
            +
            {data.commonOps.taxableProfit.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Prejuízo a compensar</p>
          <p>
            -
            {data.commonOps.compensatedLoss.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="line" />
        <div className="taxes">
          <p>Base de Cálculo</p>
          <p>
            =
            {data.commonOps.calcBase.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="result">
          <p>Imposto devido</p>
          <p>
            =
            {data.commonOps.taxDue.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
      </div>
      )}

      {data.dayTrade && (
      <div className="operationType">
        <h2>Day Trade</h2>
        <div className="taxes">
          <p>Resultado do mês</p>
          <p>
            {(data.dayTrade.taxableProfit >= 0) && '+'}
            {data.dayTrade.taxableProfit.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Prejuízo a compensar</p>
          <p>
            -
            {data.dayTrade.compensatedLoss.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="line" />
        <div className="taxes">
          <p>Base de Cálculo</p>
          <p>
            =
            {data.dayTrade.calcBase.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="result">
          <p>Imposto devido</p>
          <p>
            =
            {data.dayTrade.taxDue.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
      </div>
      )}

      {data.fii && (
      <div className="operationType">
        <h2>FII</h2>
        <div className="taxes">
          <p>Resultado Líquido</p>
          <p>
            {(data.fii.liquidResult >= 0) && '+'}
            {data.fii.liquidResult.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Prejuízo a compensar</p>
          <p>
            -
            {data.fii.compensatedLoss.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="line" />
        <div className="taxes">
          <p>Base de Cálculo</p>
          <p>
            =
            {data.fii.calcBase.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="result">
          <p>Imposto devido</p>
          <p>
            =
            {data.fii.taxDue.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
      </div>
      )}

      {data.finalResults && (
      <div className="operationType">
        <h2>Resultados Finais</h2>
        <div className="taxes">
          <p>Imposto Operação Comum</p>
          <p>
            +
            {data.finalResults.commonOpTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Imposto Day Trade</p>
          <p>
            +
            {data.finalResults.daytradeTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Imposto retido na fonte</p>
          <p>
            -
            {data.finalResults.withholdingTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="line" />
        <div className="taxes">
          <p>Imposto Total da Bolsa</p>
          <p>
            =
            {data.finalResults.totalTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>Imposto FII</p>
          <p>
            +
            {data.finalResults.fiiTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="taxes">
          <p>DARFs pendentes</p>
          <p>
            +
            {data.finalResults.previousDarf.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
        <div className="result">
          <p>Imposto a pagar</p>
          <p>
            =
            {data.finalResults.payableTax.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
          </p>
        </div>
      </div>
      )}
    </ModalDetails>
  );
}
