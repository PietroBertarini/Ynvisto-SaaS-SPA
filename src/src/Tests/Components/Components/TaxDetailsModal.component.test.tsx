import React from 'react';
import { shallow } from 'enzyme';
import TaxDetailsModal from '../../../Components/Components/TaxDetailsModal/TaxDetailsModal.component';
import {
  taxDetailsModalInfoAllData,
  taxDetailsModalInfoWithoutCommonOps,
  taxDetailsModalInfoWithoutData,
  taxDetailsModalInfoOnlyDayTrade,
} from '../../../Mocks/TaxDetailsModal.mock';

test('Renders TaxDetailsModal with full data components and match to previously snapshot', () => {
  const component = shallow(
    <TaxDetailsModal data={taxDetailsModalInfoAllData} />,
  );
  expect(component).toMatchSnapshot();
});

test('Renders TaxDetailsModal without one field components and match to previously snapshot', () => {
  const component = shallow(
    <TaxDetailsModal data={taxDetailsModalInfoWithoutCommonOps} />,
  );
  expect(component).toMatchSnapshot();
});

test('Renders TaxDetailsModal only with final results components and match to previously snapshot', () => {
  const component = shallow(
    <TaxDetailsModal data={taxDetailsModalInfoWithoutData} />,
  );
  expect(component).toMatchSnapshot();
});

test('Renders TaxDetailsModal with only one field components and match to previously snapshot', () => {
  const component = shallow(
    <TaxDetailsModal data={taxDetailsModalInfoOnlyDayTrade} />,
  );
  expect(component).toMatchSnapshot();
});
