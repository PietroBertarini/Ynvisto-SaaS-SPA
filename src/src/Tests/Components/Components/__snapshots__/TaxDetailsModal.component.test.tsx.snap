// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TaxDetailsModal only with final results components and match to previously snapshot 1`] = `
<styled.div>
  <h1>
    Detalhamento
  </h1>
  <div
    className="operationType"
  >
    <h2>
      Resultados Finais
    </h2>
    <div
      className="taxes"
    >
      <p>
        Imposto Operação Comum
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto Day Trade
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto retido na fonte
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Imposto Total da Bolsa
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto FII
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        DARFs pendentes
      </p>
      <p>
        +
        R$ 188,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto a pagar
      </p>
      <p>
        =
        R$ 188,00
      </p>
    </div>
  </div>
</styled.div>
`;

exports[`Renders TaxDetailsModal with full data components and match to previously snapshot 1`] = `
<styled.div>
  <h1>
    Detalhamento
  </h1>
  <div
    className="operationType"
  >
    <h2>
      Operações Comuns
    </h2>
    <div
      className="taxes"
    >
      <p>
        Venda total do mês
      </p>
      <p>
        R$ 10.000,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Resultado do mês
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Lucro tributável
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      Day Trade
    </h2>
    <div
      className="taxes"
    >
      <p>
        Resultado do mês
      </p>
      <p>
        +
        R$ 100,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      FII
    </h2>
    <div
      className="taxes"
    >
      <p>
        Resultado Líquido
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 87,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      Resultados Finais
    </h2>
    <div
      className="taxes"
    >
      <p>
        Imposto Operação Comum
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto Day Trade
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto retido na fonte
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Imposto Total da Bolsa
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto FII
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        DARFs pendentes
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto a pagar
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
</styled.div>
`;

exports[`Renders TaxDetailsModal with only one field components and match to previously snapshot 1`] = `
<styled.div>
  <h1>
    Detalhamento
  </h1>
  <div
    className="operationType"
  >
    <h2>
      Day Trade
    </h2>
    <div
      className="taxes"
    >
      <p>
        Resultado do mês
      </p>
      <p>
        +
        R$ 65,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      Resultados Finais
    </h2>
    <div
      className="taxes"
    >
      <p>
        Imposto Operação Comum
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto Day Trade
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto retido na fonte
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Imposto Total da Bolsa
      </p>
      <p>
        =
        R$ 98,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto FII
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        DARFs pendentes
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto a pagar
      </p>
      <p>
        =
        R$ 98,00
      </p>
    </div>
  </div>
</styled.div>
`;

exports[`Renders TaxDetailsModal without one field components and match to previously snapshot 1`] = `
<styled.div>
  <h1>
    Detalhamento
  </h1>
  <div
    className="operationType"
  >
    <h2>
      Day Trade
    </h2>
    <div
      className="taxes"
    >
      <p>
        Resultado do mês
      </p>
      <p>
        +
        R$ 100,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      FII
    </h2>
    <div
      className="taxes"
    >
      <p>
        Resultado Líquido
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Prejuízo a compensar
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Base de Cálculo
      </p>
      <p>
        =
        R$ 87,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto devido
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
  <div
    className="operationType"
  >
    <h2>
      Resultados Finais
    </h2>
    <div
      className="taxes"
    >
      <p>
        Imposto Operação Comum
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto Day Trade
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto retido na fonte
      </p>
      <p>
        -
        R$ 0,00
      </p>
    </div>
    <div
      className="line"
    />
    <div
      className="taxes"
    >
      <p>
        Imposto Total da Bolsa
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        Imposto FII
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="taxes"
    >
      <p>
        DARFs pendentes
      </p>
      <p>
        +
        R$ 0,00
      </p>
    </div>
    <div
      className="result"
    >
      <p>
        Imposto a pagar
      </p>
      <p>
        =
        R$ 0,00
      </p>
    </div>
  </div>
</styled.div>
`;
